pkg/splunk/enterprise/util.go

```diff
@@ -993,9 +1040,164 @@ func initAndCheckAppInfoStatus(client splcommon.ControllerClient, cr splcommon.M
 			appStatusContext.AppFrameworkConfig = *appFrameworkConf
 		}
 
-		// set the last check time to current time
-		SetLastAppInfoCheckTime(appStatusContext)
+		// set the last check time to current time only if the polling is enabled
+		if isAppRepoPollingEnabled(appStatusContext) {
+			SetLastAppInfoCheckTime(appStatusContext)
+		} else {
+			var status string
+			configMapName := GetSplunkManualAppUpdateConfigMapName()
+			namespacedName := types.NamespacedName{Namespace: cr.GetNamespace(), Name: configMapName}
+			configMap, err := splctrl.GetConfigMap(client, namespacedName)
+			if err != nil {
+				scopedLog.Error(err, "Unable to get configMap", "name", namespacedName.Name)
+				return err
+			}
+
+			// reset the LastAppInfoCheckTime to 0 so that we don't reconcile again and poll for apps status
+			appStatusContext.LastAppInfoCheckTime = 0
+
+			numOfObjects := getManualUpdateRefCount(client, cr, configMapName)
+
+			// turn off the manual checking for this CR kind in the configMap
+			if turnOffManualChecking == true {
+				scopedLog.Info("Turning off manual checking of apps update", "Kind", kind)
+				// reset the status back to "off" and
+				// refCount to original count
+				status = "off"
+				numOfObjects = getNumOfOwnerRefsKind(configMap, kind)
```

Comment:
I may be missing something, please correct me if my understanding is wrong. Are we prematurely marking it as `off`, in the following case?:

1. Assume there are 4 standalone CRs, S1 to S4.
2. Admin sets the flag `on` in the configMap
3. CR for S1 kicks in and sets the refCount to 1
4. Now, immediately sets the flag to `off`, as the `turnOffManualChecking` becomes true?
5. So, S2, to S4 doesn't get their turns?

Reply:
ConfigMap is ONLY created when a CR(with app framework) is applied for the first time. Only then can an admin update/change the configMap. So you need to swap step 2 and 3